plugins/chain_api_plugin/chain_api_plugin.cpp

```diff
@@ -41,6 +41,48 @@ parse_params<chain_apis::read_only::get_transaction_status_params, http_params_t
    }
 }
 
+// if actions.data & actions.hex_data provided, use the hex_data since only currently support unexploded data
+template<>
+chain_apis::read_only::get_transaction_id_params
+parse_params<chain_apis::read_only::get_transaction_id_params, http_params_types::params_required>(const std::string& body) {
+   if (body.empty()) {
+      EOS_THROW(chain::invalid_http_request, "A Request body is required");
+   }
+
+   try {
+      fc::variant trx_var =  fc::json::from_string( body );
+      if( trx_var.is_object() ) {
+         fc::variant_object& vo = trx_var.get_object();
+         if( vo.contains("actions") && vo["actions"].is_array() ) {
+            fc::mutable_variant_object mvo = vo;
+            fc::variants& action_variants = mvo["actions"].get_array();
+            for( auto& action_v : action_variants ) {
+               if( action_v.is_object() ) {
```

Comment:
cleos returns an error when one of the provided actions is not an object. Should we do the same instead of just skipping  it?

Reply:
Thank you, added (2 more exceptions from cleos)! That will definitely make api replies much more useful.